### ISSUE TITLE
Failed assertion with AMG (ep->dci == UCT_DC_EP_NO_DCI)

### ISSUE THREAD

**Issue by brminich** [openucx/ucx#2490]
The following assertion happens on ARM with AMG (problem 2):
`dc_ep.c:136  Assertion `ep->dci == UCT_DC_EP_NO_DCI' failed`

When running with release version (no assertions present), the following error arises:
```
[hpc-arm-hp01:87219:0:87219] ib_mlx5_log.c:113  FATAL: Error on QP 0x24125 wqe[1338]: Transport retry count exceeded (synd 0x15 vend 0x81) opcode SEND
```
Cmd line is:
```
mpirun -np $NP ${MPI_FLAGS} ${UCX_FLAGS} ${HCOLL_FLAGS} ./amg -problem 2 -n 40 40 40 -P 7 6 4
```
Where:
```
MPI_FLAGS="--map-by node:PE=${THREADS} --rank-by core --report-bindings --bind-to core"
UCX_FLAGS="-mca pml ucx -x UCX_NET_DEVICES=mlx5_0:1 -mca btl_openib_if_include mlx5_0:1"
HCOLL_FLAGS="-mca coll_fca_enable 0 -mca coll_hcoll_enable 1 -x HCOLL_MAIN_IB=mlx5_0:1"
```

**Comment by gcstoianowski** [comment]
Note:  The value of $NP must be equal to the product of the three integers that follow "-P".  In the above example, NP would have to be 7*6*4=168.  (In other words, -P specifies the decomposition.)

**Comment by yosefe** [comment]
@brminich can we close this?

**Comment by brminich** [comment]
fixed by #2740
